### DISCUSSION TITLE
Support list of lists arguments in dagger call

### DISCUSSION THREAD

**Discussion by Dudesons** [dagger/dagger#12347] [Help]
### What is the issue?

When a function has an optional argument with a list of lists, the function is callable from a programmatic way but from the cli an error is thrown.

Link to discord [discussion](https://discord.com/channels/707636530424053791/1120503349599543376/1214729069783023706)

### Dagger version

dagger v0.10.0 (registry.dagger.io/engine) linux/amd64

### Steps to reproduce

You need to have in your module a function with an optional list of lists like that:
```go
type MyModule struct {}

func (m *MyModule) MyFunction(
	// This argument break the cli
	// +optional
	systemSetupCmds [][]string,
) (*Node, error) {
	return m, nil
}
```

Then call this one without referencing the optional argument: `dagger call my-function`

### Log output

https://gist.github.com/Dudesons/4b0c40c5dfafa7661fa3946a36a38234

**Comment by helderco** [comment]
This isn't a bug, it's just not supported so it's a feature request:
```
Error: unsupported list of lists for flag: system-setup-cmds
```

**Comment by helderco** [comment]
What would be the syntax for setting a list of lists in the CLI?

**Comment by TomChv** [comment]
We now have a way to send list arguments: https://docs.dagger.io/api/arguments/#array-arguments

I'm closing this issue but feel free to reopen it if there's still a problem around list arg

**Comment by helderco** [comment]
We have lists of primitives, not list of lists.
